pkg/kn/commands/service/service_create.go

```diff
@@ -131,17 +131,27 @@ func createService(client serving_v1alpha1_client.ServingV1alpha1Interface, serv
 }
 
 func replaceService(client serving_v1alpha1_client.ServingV1alpha1Interface, service *serving_v1alpha1_api.Service, namespace string, out io.Writer) error {
-	existingService, err := client.Services(namespace).Get(service.Name, v1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	service.ResourceVersion = existingService.ResourceVersion
-	_, err = client.Services(namespace).Update(service)
-	if err != nil {
-		return err
+	var retries = 0
+	for true {
+		existingService, err := client.Services(namespace).Get(service.Name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		service.ResourceVersion = existingService.ResourceVersion
+		_, err = client.Services(namespace).Update(service)
+		if err != nil {
+			// Retry to update when a resource version conflict exists
+			if api_errors.IsConflict(err) && retries < MaxUpdateRetries {
```

Comment:
should we add a message here about conflict occurred and we're retrying ? 
this is just to let user know about provision we have in client for retrying
WDYT @rhuss ?

Reply:
I don't think its needed as the outcome will be the same. For `kn service create --force` we are unconditionally overwriting the service anyway, regardless what it contains on the server side. So there is no visible difference for the user.

For `kn service update` we are also refetching the current object and only apply our changes also on second and third retry.

Doing a retry is mostly an implementation details a user should not worry about.

Reply:
alright.